### ISSUE TITLE
GradNorm on a subset of loss functions

### ISSUE THREAD

**Issue by AndrewUlmer** [lucidrains/gradnorm-pytorch#11]
Maybe this shouldn't go in issues. Say I have several loss functions, is there a straightforward way to use this package to apply GradNorm only to a subset of those losses?

**Comment by lucidrains** [comment]
@AndrewUlmer not yet, but i can build this! (specify which losses get dynamic vs static scale)

**Comment by lucidrains** [comment]
@AndrewUlmer is it working well for you otherwise?

**Comment by AndrewUlmer** [comment]
That would be awesome, and yes it's working well! I can also give it a try myself.

**Comment by lucidrains** [comment]
@AndrewUlmer your letting me know it works is actually very helpful! (Most research don't work) Will get it built tomorrow!

**Comment by AndrewUlmer** [comment]
I should clarify then - I _think_ it's working 🙂

**Comment by lucidrains** [comment]
there's a frozen flag there for you to test with and without, and you can answer that question

**Comment by lucidrains** [comment]
@AndrewUlmer want to try [this](https://github.com/lucidrains/gradnorm-pytorch/commit/0eec65631920c0668bd46b7e40679e8b0084cef1#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R116) and see if it works?

**Comment by AndrewUlmer** [comment]
Yeah, I'll give it a try tomorrow! Thanks!

**Comment by AndrewUlmer** [comment]
I think [this line](https://github.com/lucidrains/gradnorm-pytorch/blob/0eec65631920c0668bd46b7e40679e8b0084cef1/gradnorm_pytorch/gradnorm_pytorch.py#L285) is not actually doing the in-place update.

**Comment by lucidrains** [comment]
@AndrewUlmer oh shoot! you are correct

should be fixed

**Comment by AndrewUlmer** [comment]
I think it's working! Thank you!
